### PR TITLE
Add tests for internal/upgrade package

### PR DESCRIPTION
**Pull request by atanasdinov**
- Adds unit tests covering the construction of different resources in the upgrade package
- Tests do include a lot of repetition and manual assertions due to the nature of the constantly changing resources

### CODE REVIEW

File: internal/upgrade/os_test.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+	require.Nil(t, upgradePlan.Spec.Prepare)
+
+	upgradeContainer := upgradePlan.Spec.Upgrade
+	require.NotNil(t, upgradeContainer)
+	assert.Equal(t, "registry.suse.com/bci/bci-base:15.5", upgradeContainer.Image)
+	assert.Equal(t, []string{"chroot", "/host"}, upgradeContainer.Command)
+	assert.Equal(t, []string{"sh", "/run/system-upgrade/secrets/some-secret/os-upgrade.sh"}, upgradeContainer.Args)
+
+	assert.Equal(t, "3.1.0", upgradePlan.Spec.Version)
```
Comment (atanasdinov) [inline internal/upgrade/os_test.go:85]: I'm not quite confident we need to specify this version, but investigating this is a very low priority at this point.

File: internal/upgrade/kubernetes_test.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+	matchExpression := upgradePlan.Spec.NodeSelector.MatchExpressions[0]
+	assert.Equal(t, "node-role.kubernetes.io/control-plane", matchExpression.Key)
+	assert.EqualValues(t, "NotIn", matchExpression.Operator)
+	assert.Equal(t, []string{"true"}, matchExpression.Values)
+
+	prepareContainer := upgradePlan.Spec.Prepare
+	require.NotNil(t, prepareContainer)
+	assert.Equal(t, "rancher/rke2-upgrade", prepareContainer.Image)
+	assert.Empty(t, prepareContainer.Command)
+	assert.Equal(t, []string{"prepare", "control-plane-v1-30-2-rke2r1-abcdef"}, prepareContainer.Args)
```
Comment (atanasdinov) [inline internal/upgrade/kubernetes_test.go:168]: Yet another thing that we might not need is the `Prepare` configuration for Kubernetes plans. We ditched the logic which creates the two plans at the same time and only reconcile worker nodes once after all control plane ones are processed. This is also why a `Prepare` configuration is missing in the OS plans too. Again, probably not worth taking the time to look at right now.

File: internal/upgrade/os_test.go (hunk 1 above)
Comment (ipetrov117) [reply internal/upgrade/os_test.go:85]: Yea, so this is specified only for a better looking output when a user does `kubectl get plans`. If the field is really required, I am not sure, but this is definitely low priority.

File: internal/upgrade/kubernetes_test.go (hunk 2 above)
Comment (ipetrov117) [reply internal/upgrade/kubernetes_test.go:168]: With the current workflow of the `upgrade-controller` we might really need to remove the `Prepare` blocks, but I agree that we should look at this post release.

**Review by ipetrov117 — APPROVED** [review]
Looks good as a first unit-test version.

**Pull request merged**